### PR TITLE
Add to docstrings in `plasmapy/formulary/magnetostatics.py`

### PR DESCRIPTION
**Pull request by TrestanSimon**
This PR fixes and adds to/rewrites the [Notes](https://numpydoc.readthedocs.io/en/latest/format.html#notes) sections of the docstrings associated with the `magnetic_field` functions within `plasmapy/formulary/magnetostatics.py`. Previously, the Notes sections were either incomplete and/or incorporated broken math directives. Here I have added to them so that the equations used (and approximation if present) are explained somewhat briefly and the style used is maintained between docstrings.

- [x] I have added a changelog entry for this pull request.

- [x] I have fixed any newly failing tests.

### CODE REVIEW

**Comment by TrestanSimon** [comment]
FYI, I began this PR prior to becoming aware of https://github.com/PlasmaPy/PlasmaPy/issues/1688#issue-1348317679. Perhaps merging should wait until a decision has been made.

File: plasmapy/formulary/magnetostatics.py [hunk 1]
```diff
@@ -250,14 +301,57 @@ def magnetic_field(self, p) -> u.T:
...
+        :math:`\vec{p}_1`, and :math:`\vec{p}_2`, respectively).
+        Further, the vector from points
+        :math:`p_i` to :math:`p_j` can be written as
+        :math:`\vec{p}_{ij} = \vec{p}_j - \vec{p}_i`.
+
+        Next, consider the triangle with the points :math:`p_0`,
+        :math:`p_1`, and :math:`p_2` as vertices. The foot of
+        perpendicular opposite the vertex :math:`p_0`, which will be
+        used to find the unit vector in the direction of the magnetic
+        field, can be expressed as
```
Comment (namurphy) [inline plasmapy/formulary/magnetostatics.py:321]: I got a little confused here what "foot of perpendicular opposite the vertex" means. Would you be able to clarify or reword it?

**Review by namurphy — COMMENTED** [review]
Thank you for doing this, @TrestanSimon!  I read through the changes, and only had a few slight suggestions and one place where I was wondering if a rewording would help. Overall this will be a significant improvement!

Even given the discussion in https://github.com/PlasmaPy/PlasmaPy/issues/1688#issue-1348317679, I think it'll still be very much worth merging, in particular since it will only require a few more changes.  We haven't officially decided to deprecate `plasmapy.magnetostatics`, and even if we did, we would probably keep it around with a deprecation warning for ∼2 releases before finally removing it.

Thank you again!

File: plasmapy/formulary/magnetostatics.py (hunk 1 above)
Comment (TrestanSimon) [reply plasmapy/formulary/magnetostatics.py]: Considering the triangle with vertices `p0`, `p1`, and `p2`, ~~the "foot of perpendicular opposite the vertex `p0`"~~ the "vector from the vertex `p0` to the perpendicular foot opposite the vertex `p0`" (amended in 88e42da) refers to the vector from the vertex `p0` to the opposite side of the triangle (the line formed between `p1` and `p2`) and perpendicular to this opposite side.

I made a slight adjustment to the wording in 88e42da making it clearer to me, but let me know if it is still unclear for you. Thanks for looking over this.

**Review by namurphy — APPROVED** [review]
Thank you again for doing this!  I'm happy to approve and merge it.

**Pull request merged**